src/python/platforms/android/settings.py

```diff
@@ -150,6 +167,36 @@ def get_security_patch_level():
   return adb.get_property('ro.build.version.security_patch')
 
 
+def get_kernel_build_id():
+  """Returns the build id of the kernel from the kernel version string."""
+  version_string = adb.run_shell_command('cat /proc/version').strip()
+  # Linux version 3.18.0-g(8de8e79)-ab(1234567) where 8de8e79 is the hash and
+  # 1234567 optional is the kernel build id.
+  match = re.match(r'Linux version .+-g([0-9a-f]+)[\s\-](ab([0-9a-f]+)\s)',
+                   version_string)
+  if match:
+    return match.group(3)
+
+  return None
+
+
+def get_kernel_name():
+  """Returns the kernel name for the device, since some kernels are shared."""
+  product_name = get_product_name()
+  build_product = get_build_product()
+
+  # Some devices have a different kernel name than product_name, if so use the
+  # kernel name, but don't forget the kasan suffix if it exists.
+  if product_name in PRODUCT_TO_KERNEL:
+    return PRODUCT_TO_KERNEL[build_product]
```

Comment:
What if device family is not in list, maybe use attribute.get() so that it returns None.